cpp/src/join/joining.cu

```diff
@@ -537,6 +539,63 @@ gdf_error join_call_compute_df(
     return GDF_SUCCESS;
   }
 
+
+  //if the inputs are nvcategory we need to make the dictionaries comparable
+  bool const at_least_one_category_column{ std::any_of( left_cols, left_cols + num_cols_to_join, [](gdf_column * col){ return col->dtype == GDF_STRING_CATEGORY; }) };
```

Comment:
This won't work as-is, and I think it's my fault because I original suggested it.

You're iterating `[left_cols, left_cols + num_cols_to_join)`, which isn't correct. 

You need to iterate only the columns we're joining on, which are indicated by `left_join_cols`. In which case, `any_of` may not work here.

Reply:
wow I really hate how that reads and looks. You c++ peeps... :)

Reply:
I did what @harrism suggeste I hope its ok :)

Reply:
@harrism's suggestion was great, but has a syntax error:

```
std::any_of( left_join_cols, left_join_cols + num_cols_to_join, [left_cols](int col){ return left_cols[col]->dtype == GDF_STRING_CATEGORY; });
```

@felipeblazing make sure to correct that if you haven't already.